include/seastar/util/noncopyable_function.hh

```diff
@@ -66,9 +67,17 @@ private:
         // that, and complains. Silence it.
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wuninitialized"
+        // A bytewise loop doesn't optimize to a wordwise loop since
+        // the compiler can't guess if there is overlap or not. Help
+        // it out by using a temporary, which the compiler knows cannot
+        // overlap.
+        std::array<char, N> tmp;
         // Avoid including <algorithm> just for this
-        for (unsigned i = 0; i != N; ++i) {
-            to->_storage.direct[i] = from->_storage.direct[i];
+        for (size_t i = 0; i != N; ++i) {
+            tmp[i] = from->_storage.direct[i];
+        }
+        for (size_t i = 0; i != N; ++i) {
+            to->_storage.direct[i] = tmp[i];
```

Comment:
And then we wonder why Seastar-based code takes hours to compile ;-)

Reply:
You are forcing the compiler to instantiate a template array<N>, compile two loops that copy data byte by byte and increment the counter i from 0 to N twice and copy each byte twice - and then run its fancy optimizer until absolutely nothing is left from all of this except one assembler instruction.

This results in excellent performance, but also terrible compilation speed.